### PR TITLE
MCKIN-24383 xblock internationalization

### PR DESCRIPTION
**Pull request by mudassir-hafeez**
In order to make a Group project v2  translatable, I have integrated translation and added a minimal number of hooks to Python code and templates. By this PR, the text should be translated into the end user’s language.

Testing instructions:
-  checkout to this branch
- and to verify the strings are now getting translated for all languages.

### CODE REVIEW

**Comment by viadanna** [comment]
Hello @mudassir-hafeez can you provide a brief description of the changes and testing instructions?

Also, there are some issues that prevent CI testes passing, can you check those?

**Comment by xitij2000** [comment]
@mudassir-hafeez Could you look at the test failures here. They seem to be due to minor isort issues. 

I'll review this shortly.

**Comment by mudassir-hafeez** [comment]
@xitij2000 fixed the isort issue. Can you please review?

**Comment by xitij2000** [comment]
@mudassir-hafeez Does this need to go only on juniper? If not, then it might make sense to also create a PR against the v0.10 branch.

File: pylintrc [hunk 1]
```diff
@@ -8,6 +8,7 @@ symbols=yes
 
 [MESSAGES CONTROL]
 disable=
+ invalid-name,
```
Comment (xitij2000) [inline pylintrc:11]: This is overly broad, remove this.

File: pylintrc [hunk 2]
```diff
@@ -19,7 +20,13 @@ disable=
...
 good-names=id,log,loader,setUp
 max-args=6
-method-rgx=_?[a-z_][a-z0-9_]{2,40}$
+
+# Naming style matching correct method names
+method-naming-style=snake_case
+
+# Regular expression matching correct method names. Overrides method-naming-
+# style,
+#method-rgx=_?[a-z_][a-z0-9_]{2,40}$
```
Comment (xitij2000) [inline pylintrc:29]: I think the options here are pretty descriptive, so no need for additional comments. 
```suggestion
method-naming-style=snake_case
```

File: setup.py [hunk 3]
```diff
@@ -25,7 +25,7 @@ def package_data(pkg, root_list):
 
 setup(
     name='xblock-group-project-v2',
-    version='0.11.0',
+    version='0.11.1',
```
Comment (xitij2000) [inline setup.py:28]: This includes pretty big changes so I suggest we bump the minor version. 
```suggestion
    version='0.12.0',
```

**Review by xitij2000 — CHANGES_REQUESTED** [review]
This seems mostly good to go, just need some minor changes.

**Comment by mudassir-hafeez** [comment]
@xitij2000 suggestions have been addressed. Here #152 is the actual PR against v0.10. Can you please review and create a release tag?

**Review by xitij2000 — APPROVED** [review]
:+1:
- [x] I tested this: tested on juniper solution devstack
- [x] I read through the code
- [na] I checked for accessibility issues
- [na] Includes documentation

**Pull request merged**